docs/proposals/local-registries.md

```diff
@@ -0,0 +1,165 @@
+---
+title: local-registries
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@coreydaley"
+  - "@otaviof"
+approvers:
+  - "@qu1queee"
+  - "@sbose78"
+creation-date: 2020-07-24
+last-updated: 2020-07-24
+status: provisional
+see-also: []  
+replaces: []
+superseded-by: []
+---
+
+# Local Registries
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+[KEP 1755](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry)
+provides an official convention for clusters to declare the presence of an internal image registry.
+This proposal aims to take advantage of declared local image registries, and let application
+developers simplify how they declare image pull specs. This may also help `Build` objects be
+portable across clusters.
+
+## Motivation
+
+The `Build` API in its present form requires that users provide a full image pull spec for image
+references. This can be cumbersome when referencing an image on an internal registry, such as those
+present in [KIND](https://kind.sigs.k8s.io/docs/user/local-registry/),
+[Rancher](https://github.com/rancher/k3d/blob/master/docs/registries.md#using-a-local-registry),
+[microk8s](https://microk8s.io/docs/registry-built-in), and
+[OpenShift](https://docs.openshift.com/container-platform/4.5/registry/architecture-component-imageregistry.html).
+This proposal will provide a means of declaring that an image spec references a local registry,
+thereby allowing host information to be removed from the image reference.
+
+### Goals
+
+- Simplify how local image registries can be referenced in `Build` objects.
+- Provide a means of declaring that an image references a cluster-local image registry.
+
+### Non-Goals
+
+- Automate how pull/push secrets can be added to a `Build` object.
+- Support secure image pull/push if a local registry uses self-signed certificates.
+
+## Proposal
+
+Application developers will be able to specify `local: true|false` if an image reference points
+to a cluster-local image registry.
+
+The build controller/operator will watch the global "LocalRegistryHosting" `ConfigMap` to check if
```

Comment:
"global" here really means there is a single instance in a given namespace, no?  

should we proscribe that the configmap lives in the namespace (or one of the namespaces) created when this feature is installed ?

Reply:
Accordingly with upstream [KEP-1755](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry#the-local-registry-hosting-configmap), "global" means a single `ConfigMap` in `kube-public` namespace.

Reply:
Correct - it is specifically `kube-public/local-registry-hosting`. This is akin to the OpenShift global configurations that live in `openshift-config` - most have canonical names.

Reply:
ah - cool, thanks for the details / clarifications

Reply:
though perhaps a reference to KEP-1755 here would help the uninitiated